exporter/awsprometheusremotewriteexporter/auth.go

```diff
@@ -100,6 +107,20 @@ func getCredsFromConfig(auth AuthConfig) *credentials.Credentials {
 	return creds
 }
 
+func parseEndpointRegion(endpoint string) (region string, err error) {
+	// Example: https://aps-workspaces.us-east-1.amazonaws.com/workspaces/ws-XXX/api/v1/remote_write
+	const nDomains = 3
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", err
+	}
+	p := strings.SplitN(u.Host, ".", nDomains)
+	if len(p) < nDomains {
+		return "", fmt.Errorf("invalid endpoint: %q", endpoint)
```

Comment:
By the way do we want this to be an err, i.e., what if taking to localhost for development? Not sure it's possible though.

Reply:
Users can set the region setting manually in those cases.